nextest-runner/src/max_fail.rs

```diff
@@ -0,0 +1,38 @@
+use std::{fmt, str::FromStr};
+
+use crate::errors::MaxFailParseError;
+
+/// Type for the max-fail flag
```

Comment:
Could you put this file inside config? That's where this usually lives (and would align with future work to make config support max-fail as well).